homeassistant/components/fritz/image.py

```diff
@@ -0,0 +1,90 @@
+"""FRITZ image integration."""
+
+from __future__ import annotations
+
+from io import BytesIO
+import logging
+
+from homeassistant.components.image import ImageEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util, slugify
+
+from .common import AvmWrapper, FritzBoxBaseEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up guest WiFi QR code for device."""
+    avm_wrapper: AvmWrapper = hass.data[DOMAIN][entry.entry_id]
+
+    guest_wifi_info = await hass.async_add_executor_job(
+        avm_wrapper.fritz_guest_wifi.get_info
+    )
+
+    if not guest_wifi_info.get("NewEnable"):
+        return
+
+    async_add_entities(
+        [
+            FritzGuestWifiQRImage(
+                avm_wrapper, entry.title, guest_wifi_info["NewSSID"], hass
+            )
+        ]
+    )
+
+
+class FritzGuestWifiQRImage(FritzBoxBaseEntity, ImageEntity):
+    """Implementation of the FritzBox guest wifi QR code image entity."""
+
+    _attr_content_type = "image/png"
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        avm_wrapper: AvmWrapper,
+        device_friendly_name: str,
+        ssid: str,
+        hass: HomeAssistant,
```

Comment:
Please always put `hass` as the first parameter when present in a signature. That's our convention.

Reply:
will create a followup PR later this day

Reply:
#95437